src/function/table/read_csv.cpp

```diff
@@ -482,17 +530,18 @@ bool ParallelCSVGlobalState::Next(ClientContext &context, const ReadCSVData &bin
 			// we are doing UNION BY NAME - fetch the options from the union reader for this file
 			auto &union_reader = *bind_data.union_readers[file_index - 1];
 			reader = make_uniq<ParallelCSVReader>(context, union_reader.options, std::move(result), first_position,
-			                                      union_reader.GetTypes());
+			                                      union_reader.GetTypes(), &line_info, file_index - 1);
 			reader->names = union_reader.GetNames();
 		} else if (file_index <= bind_data.column_info.size()) {
 			// Serialized Union By name
-			reader = make_uniq<ParallelCSVReader>(context, bind_data.options, std::move(result), first_position,
-			                                      bind_data.column_info[file_index - 1].types);
+			reader =
+			    make_uniq<ParallelCSVReader>(context, bind_data.options, std::move(result), first_position,
+			                                 bind_data.column_info[file_index - 1].types, &line_info, file_index - 1);
 			reader->names = bind_data.column_info[file_index - 1].names;
 		} else {
 			// regular file - use the standard options
 			reader = make_uniq<ParallelCSVReader>(context, bind_data.options, std::move(result), first_position,
-			                                      bind_data.csv_types);
+			                                      bind_data.csv_types, &line_info, file_index - 1);
```

Comment:
I'm guessing the LineInfo is kept per-file - does it make sense to move this to the `CSVBufferRead` instead of the parallel CSV state?